specs/TrustedOrigin.md

```diff
@@ -0,0 +1,213 @@
+  HRESULT GetOriginSettings(
+      [out, retval] ICoreWebView2StagingOriginSettingCollection** value);
+
+  /// Creates a new `OriginSetting` object.
+  HRESULT CreateOriginSetting(
+      [out, retval] ICoreWebView2StagingOriginSetting** value);
+}
+
+/// Represents settings for a specific origin.
+interface ICoreWebView2StagingOriginSetting : IUnknown {
+  /// Gets the `AccentColor` property.
+  [propget] HRESULT AccentColor([out, retval] OriginSettingState* value);
+
+
+  /// Sets the enabled property for Accent Color for the associated origin.
+  /// The default value for this is `Block`. When set to `Allow`, the origin is
+  /// allowed to use AccentColor keyword or accent-color CSS property to display OS Accent Color.
+  [propput] HRESULT AccentColor([in] OriginSettingState value);
+
+
+  /// Gets the `EnhancedSecurityMode` property.
+  [propget] HRESULT EnhancedSecurityMode([out, retval] OriginSettingState* value);
+
+
+  /// Sets the enabled property for Enhanced Security Mode for the associated origin.
+  /// When set to `Block`, the origin bypasses Enhanced Security Mode even if
+  /// Enhanced Security Mode is enabled for the profile. Similarly, when set to `Allow`,
+  /// the origin enforces Enhanced Security Mode even if Enhanced Security Mode is disabled
+  /// for the profile.
+  [propput] HRESULT EnhancedSecurityMode([in] OriginSettingState value);
+
+
+  /// Gets the `OriginPattern` property.
+  ///
+  /// The caller must free the returned string with `CoTaskMemFree`.  See
+  /// [API Conventions](/microsoft-edge/webview2/concepts/win32-api-conventions#strings).
+  [propget] HRESULT OriginPattern([out, retval] LPWSTR* value);
+
+
+  /// Sets the origin pattern associated with this origin setting.
+  /// The origins pattern can be both exact origin strings and wildcard patterns.
+  /// For wildcard patterns, `*` matches zero or more characters.
+  /// Examples:
+  /// | Origin Filter String | What It Matches | Description |
+  /// |---------|-----------------|-------------|
+  /// | `https://contoso.com` | Only `https://contoso.com` | Matches the exact origin with specific protocol and hostname |
+  /// | `https://*.contoso.com` | `https://app.contoso.com`,`https://api.contoso.com`,`https://admin.contoso.com` | Matches any subdomain under the specified domain |
+  /// | `*://contoso.com` | `https://contoso.com`,`http://contoso.com`,`ftp://contoso.com` | Matches any protocol for the specified hostname |
+  /// | `*contoso.*` | `https://www.contoso.com`,`http://app.contoso.com` | Matches any protocol and any subdomain under the hostname |
+  /// | `*example/` | `https://app.example/`,`https://api.example/` | Matches any subdomain and top-level domain variations |
+  /// | `https://xn--qei.example/` | `https://â¤.example/`,`https://xn--qei.example/` | Normalized punycode matches with corresponding Non-ASCII hostnames |
+  /// 
+  /// Note: `*` is not a valid value for OriginPattern.

```

Comment:
Why not? This seems like it should be valid.

Reply:
>And if there is a need for toggle control, we'll add APIs as required.

Is that the right path though? Maybe more of a question for @shrinaths / @david-risney but if we had this API (with just `*` allowed) when we were implementing something like `CoreWebView2Profile.PreferredTrackingPreventionLevel`, would we have implemented it? It seems like it could be cleaner and easier to have this API as the primary way to support enabling/disabling features for all origins in a profile for future additions.